memif-attach/main.go

```diff
@@ -18,61 +18,112 @@ package main
 
 import (
 	"context"
+	"net"
 	"time"
-
+	// "io"
+	"encoding/binary"
+	"bufio"
+	"bytes"
 	"git.fd.io/govpp.git/api"
 	"github.com/edwarnicke/govpp/binapi/memif"
+	interfaces "github.com/edwarnicke/govpp/binapi/interface"
+	"github.com/edwarnicke/govpp/binapi/interface_types"
 	"github.com/edwarnicke/log"
 	"github.com/edwarnicke/vpphelper"
 )
 
+type MemifMsg struct {
+	HelloMsgType HelloMsg
+	InitMsgType InitMsg
+}
+// HelloReplyMsg type
+type HelloMsg struct {
+	Name [32]byte // 32 bytes array
+	Min_version uint16 
+	Max_version uint16
+	Max_region uint16 
+	Max_m2s_ring uint16 
+	Max_s2m_ring uint16
+	Max_log2_ring_size uint8 
+} 
+
+// InitMsg
+const MEMIF_SECRET_SIZE = 24
+type MemifInterfaceMode int
+const (
+	MEMIF_INTERFACE_MODE_ETHERNET MemifInterfaceMode = iota
+	MEMIF_INTERFACE_MODE_IP
+	MEMIF_INTERFACE_MODE_PUNT_INJECT
+)
+type InitMsg struct {
+	Version uint16 // check the file
```

Comment:
[memif_msg_enq_init](https://github.com/FDio/vpp/blob/4b9246ad2008fa791405a3482914960a352237c9/src/plugins/memif/socket.c#L115-L130) shows you how vpp enqueues the init message when it is itself is in the slave role.

From that you will note that it sets 
```c
 i->version = MEMIF_VERSION;
```

and you can find the definition for MEMIF_VERSION [here](https://github.com/FDio/vpp/blob/4b9246ad2008fa791405a3482914960a352237c9/src/plugins/memif/memif.h#L28)

Reply:
Add Type uint16 at beginning here